src/nvim/screen.c

```diff
@@ -5394,18 +5394,18 @@ void grid_puts_len(ScreenGrid *grid, char_u *text, int textlen, int row,
   }
 
   if (do_flush) {
-    screen_puts_line_flush(true);
+    grid_puts_line_flush(grid, true);
   }
 }
 
-void screen_puts_line_flush(bool set_cursor)
+void grid_puts_line_flush(ScreenGrid *grid, bool set_cursor)
```

Comment:
Passing grid in `grid_puts_line_start` would mean creating another static global variable because we need `grid` in `grid_puts_line_flush` for `ui_line`. Or we can just pass the grid in both cases and assert that the grid is same?